scripts.js

```diff
@@ -65,6 +56,20 @@ function displayCurrentPalette() {
   }
 }
 
+function lockColor(event) {
```

Comment:
Nick did give us a hint that led us to: currentPalette[`color${i+1}`].locked = !currentPalette[`color${i+1}`].locked